commit db894649e34abceade484c35e5acd346e74e916d
Author: Hui Wang <hui.wang@canonical.com>
Date:   Fri Mar 11 12:04:02 2016 +0800

    ALSA: hda - fix the mic mute button and led problem for a Lenovo AIO
    
    commit 6ef2f68fa38bf415830f67903d87180d933e0f47 upstream.
    
    This Lenovo ThinkCentre AIO also uses Line2 as mic mute button and
    uses GPIO2 to control the mic mute led, so applying this quirk can
    make both the button and led work.
    
    BugLink: https://bugs.launchpad.net/bugs/1555912
    Signed-off-by: Hui Wang <hui.wang@canonical.com>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index c2430b3..6968b79 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -5529,6 +5529,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
 	SND_PCI_QUIRK(0x17aa, 0x2226, "ThinkPad X250", ALC292_FIXUP_TPT440_DOCK),
 	SND_PCI_QUIRK(0x17aa, 0x2233, "Thinkpad", ALC293_FIXUP_LENOVO_SPK_NOISE),
 	SND_PCI_QUIRK(0x17aa, 0x30bb, "ThinkCentre AIO", ALC233_FIXUP_LENOVO_LINE2_MIC_HOTKEY),
+	SND_PCI_QUIRK(0x17aa, 0x30e2, "ThinkCentre AIO", ALC233_FIXUP_LENOVO_LINE2_MIC_HOTKEY),
 	SND_PCI_QUIRK(0x17aa, 0x3902, "Lenovo E50-80", ALC269_FIXUP_DMIC_THINKPAD_ACPI),
 	SND_PCI_QUIRK(0x17aa, 0x3977, "IdeaPad S210", ALC283_FIXUP_INT_MIC),
 	SND_PCI_QUIRK(0x17aa, 0x3978, "IdeaPad Y410P", ALC269_FIXUP_NO_SHUTUP),
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1354_db894649e34abceade484c35e5acd346e74e916d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1354_db894649e34abceade484c35e5acd346e74e916d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
